### ISSUE TITLE
basicConstraints missing from openssl.cnf

### ISSUE THREAD

**Issue by mwasilew** [foundriesio/estserver#11]
After creating a certificate with `contrib/mk-tls-keypair.sh` script, upload to the factory failed with the following message:
```
ERROR: HTTP error during PATCH 'https://api.foundries.io/ota/factories/lmp-ci-testing/certs/': 400 BAD REQUEST
= Invalid certs: Missing required certificate extension: <ObjectIdentifier(oid=2.5.29.19, name=basicConstraints)>
```

Adding the following line to the openssl.cnf should fix the issue
```
basicConstraints=critical,CA:true,pathlen:0
```

**Comment by mwasilew** [comment]
This is what I read from README:
```
TLS certificates for the server can be generate by using the helper script
`contrib/mk-tls-keypair.sh`. Your factory's PKI directory  was generated
with a `create_ca` script. Once a CA is created, you can upload/authorize
it with:

``bash
fioctl keys ca show --just-device-cas > /tmp/cas.pem
cat <new-ca.pem> >> /tmp/cas.pem
fioctl keys ca update /tmp/cas.pem
fioctl keys ca show --just-device-cas > client-cas.pem
``
```
Is the "new-ca.pem" coming from `mk-tls-keypair.sh`? Or is it something completely different?

**Comment by doanac** [comment]
`create_ca` creates the intermediate CA you need for an EST server to sign things
`contrib/mk-tls-keypair.sh` creates the TLS cert you need for serving HTTPS to devices in a way that they'll trust you.

**Comment by mwasilew** [comment]
This is addressed in #13
